### PR TITLE
osd: Add config option osd_read_ec_check_for_errors for testing

### PR DESCRIPTION
**Pull request by dzafman**
All rados suites involving erasure codes should set this option

Signed-off-by: David Zafman <dzafman@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
@athanatos @dzafman what's the status of this pr?

**Comment by dzafman** [comment]
@liewegas @athanatos I didn't want a "Bad hash" to go undetected because we don't return errors if enough EC shards are readable.  Now that I think about it assuming that it would be caught by the log scan after a run, we don't need this change.

**Comment by dzafman** [comment]
@liewegas I think we should merge this post-Infernalis even if we don't end up using the option as long as it doesn't break things when turned off (the default).

File: src/osd/ECBackend.cc [hunk 1]
```diff
@@ -1106,11 +1106,18 @@ void ECBackend::handle_sub_read_reply(
...
-	++is_complete;
-	rop.complete[iter->first].errors.clear();
         assert(rop.complete[iter->first].r == 0);
+	if (!rop.complete[iter->first].errors.empty()) {
+	  if (cct->_conf->osd_read_ec_check_for_errors) {
+	    dout(10) << __func__ << ": Not ignoring errors, use one shard err=" << err << dendl;
+	    err = rop.complete[iter->first].errors.begin()->second;
+            rop.complete[iter->first].r = err;
+	  } else {
+	    dout(10) << __func__ << " Enough copies for " << iter->first << " (ignore errors)" << dendl;
```
Comment (liewegas) [inline src/osd/ECBackend.cc]: should this log something to the central cluster log?

**Comment by liewegas** [comment]
passed wip-sage-testing.  seems like we should also log to the cluster log if we see a disk error (even though we compensate)?

File: src/osd/ECBackend.cc (hunk 1 above)
Comment (dzafman) [inline src/osd/ECBackend.cc]: Good idea!

**Comment by dzafman** [comment]
@liewegas Added cluster logging and improved the message.  Will retest with my other pull request.

**Comment by dzafman** [comment]
@liewegas Passed a subset of rados suite (cluster logging added previously)

**Comment by dzafman** [comment]
@liewegas ping...ready to merge

**Comment by dzafman** [comment]
@athanatos Ping, ready to merge after review.

**Pull request merged**